sdk/src/index.ts

```diff
@@ -115,7 +115,7 @@ export interface MixinDeviceOptions<T> {
   constructor(options: MixinDeviceOptions<T>) {
     super();
 
-    this.nativeId = systemManager.getDeviceById(this.id)?.nativeId;
+    this.nativeId = systemManager.getDeviceById(this.id!)?.nativeId;
```

Comment:
I'm manually asserting `this.id` is not undefined, but I'm not sure this should actually be changed in the "original types".

But the original types is a bit complicated because there's multiple layers

Firstly it's optional in `types.input.ts` https://github.com/koush/scrypted/blob/8fa0da7bd812c7badbc6c2364cfbe434e491e98a/sdk/types/src/types.input.ts#L30

Then your custom build script merges together all the properties and makes it optional anyway https://github.com/koush/scrypted/blob/0b6d61a801fbe559682e0ddd439d0f257583ba36/sdk/types/src/build.ts#L43

However I think we could change the build script to check the `isOptional` flag in `schema.json` and correctly write out `:` or `?:` depending on the actual property.

```json
		{
			"id": 2,
			"name": "ScryptedDevice",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"comment": {
				"summary": [
					{
						"kind": "text",
						"text": "All devices in Scrypted implement ScryptedDevice, which contains the id, name, and type. Add listeners to subscribe to events from that device."
					}
				]
			},
			"children": [
				{
					"id": 21,
					"name": "id",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
```